NGitLab/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+
+namespace NGitLab.Extensions;
+
+internal static class TypeExtensions
+{
+    public static string GetEnumMemberAttributeValue<TEnum>(this TEnum value)
```

Comment:
I also had to add this extension method to support getting the `EnumMemberAttribute` value from the enum value for the file extension translation.

Reply:
I was sure `Utils.AddParameter` already handled the `EnumMemberAttribute`, but I was wrong.

I made the required changes in https://github.com/ubisoft/NGitLab/pull/660.

Reply:
I've rebased from main and made some further changes. 

I am unable to use `Utils.AddParameter` in this instance because in the case of the archive endpoint, the requested extension needs to be appended directly to the path when a specific format is required, i.e.:
- `/archive` - when requesting the default archive format (currently defaults to *tar.gz*)
- `/archive.ext` - when requesting a specific format where `ext` is one of the GitLab supported formats

To cater for this I've added a new method `Utils.AppendSegmentToUrl` which can be used to append a segment to the URL. I've also added a suite of unit tests for this new method. 

Please let me know if there are any issues or further changes you like me to make.